### ISSUE TITLE
Spies should return null if method is not mocked and return type is nullable

### ISSUE THREAD

**Issue by asprega** [mockery/mockery#725]
Hi. First of all, sorry if this is a duplicated issue. I tried looking for a similar one but didn't find it.

I'm on 0.9.9 stable, and I noticed a surprising behavior that I wanted to get to your attention. I'll go with an example:

```
// This is the interface I want to mock
interface Foo
{
    public function getNullableBar(): ?Bar
}
```

```
// This is the test code
$foo = \Mockery::spy(Foo::class);

$this->assertNull($foo->getNullableBar());
```

I would expect this to pass, but it fails because mockery creates a mock for `Bar` arbitrarily, even though the type is nullable and a `null` would satisfy it. So I'm always forced to add an explicit `shouldReceive->andReturnNull` for every nullable method for which I need to return null. In my opinion, this should really be the default behavior.

If maintainers agree that this is desirable, I'd be very happy to go ahead and submit a pull request. Thanks!

**Comment by davedevelopment** [comment]
I think it would be desirable, please send that PR :)

**Comment by asprega** [comment]
Here it is: https://github.com/mockery/mockery/pull/727

Let me know if I need to change anything. Thanks :)

**Comment by davedevelopment** [comment]
Fixed by #727
